ethicml/algorithms/postprocess/hardt.py

```diff
@@ -144,7 +144,7 @@ def _predict(
         test_preds_numpy: np.ndarray = test_predictions.hard.to_numpy()
 
         # Randomly flip labels according to the probabilities in model_params
-        self_fair_pred = test_preds_numpy[test.s[test.s.columns[0]].to_numpy() == 0].copy()
+        self_fair_pred = test_preds_numpy[mask_s1].copy()
```

Comment:
btw, how did you come across this?

Reply:
I have no recollection of changing this. If I had to guess, mypy not liking it?

Reply:
oh wait, it's actually wrong. it should be `mask_s0` instead of `mask_s1`, right?

Reply:
then why did the tests pass...?

Reply:
wow... that was a silly mistake... Sorry about that.
I guess there's 2 things happening.
1) Some labels in a group are still being flipped, they're just the other group
2) The tests probably didn't pass, but this was attributed to the data changing as opposed to the model

Reply:
so, I've just reverted this, and it doesn't run ...

```
self_fair_pred[p2n_indices] = self._unfavorable_label
IndexError: index 39 is out of bounds for axis 0 with size 38
```